java/fury-core/src/main/java/org/apache/fury/serializer/collection/CollectionSerializers.java

```diff
@@ -221,6 +249,22 @@ public T newCollection(MemoryBuffer buffer) {
       fury.getRefResolver().reference(collection);
       return collection;
     }
+
+    @Override
+    public Collection newCollection(Collection originCollection) {
+      Collection collection;
+      Comparator comparator = ((SortedSet) originCollection).comparator();
+      if (Objects.equals(type, TreeSet.class)) {
```

Comment:
I noticed we used `Objects` in many places, could we remove such usage, we need to reduce callstack and the size of method bytecode